lib/resources/key_rsa.rb

```diff
@@ -0,0 +1,42 @@
+require 'openssl'
+
+module Inspec::Resources
+  class RSA_Key < Inspec.resource(1)
+    name 'key_rsa'
+    desc 'public/private RSA key pair test'
+    example "
+      describe rsa_key('key.pem', 'passphrase') do
+        it { should be_valid }
+        it { should be_private }
+        it { should be_public }
+      end
+    "
+
+    def initialize(filename, passphrase = '')
+      file = inspec.file(filename)
+      @passphrase = passphrase
+      return if file.nil? || file.content.nil?
+
+      begin
+        @key = OpenSSL::PKey.read(file.content, @passphrase)
+      rescue OpenSSL::PKey::RSAError
+        # TODO: we should throw an exception here
+        return skip_resource 'Unable to load private key'
+      end
+    end
+
+    def valid?
+      public?
+    end
+
+    def public?
+      return if @key.nil?
+      @key.public?
+    end
+
+    def private?
+      return if @key.nil?
+      @key.private?
+    end
```

Comment:
So are we saying that a certificate is valid if it is public? I think we may need to refine what we mean by 'valid' just so it is clear. In the DoD sense, a key is 'valid' is it was issues and signed by an approved known trusted CA - i.e. one of the valid root or intermediate authorities.